### PR TITLE
infoschema: Add new columns for information_schema.tiflash_tables

### PR DESCRIPTION
**Pull request by JaySon-Huang**
Signed-off-by: JaySon-Huang <tshent@qq.com>

### What problem does this PR solve?

Issue Number: related to https://github.com/pingcap/tics/pull/2229

Problem Summary: Suspicious that some TiFlash tasks hold snapshot for a long time, prevent PageStorage from GCing old PageFiles. We add some information about the oldest snapshot in TiFlash system tables and want to access them by TiDB system table `information_schema.tiflash_tables`.

### What is changed and how it works?

Add column `STORAGE_{STABLE,DELTA,META}_OLDEST_SNAPSHOT_LIFETIME` and `STORAGE_{STABLE,DELTA,META}_OLDEST_SNAPSHOT_THREAD_ID` for `information_schema.tiflash_tables`

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch: 4.0, 5.0, 5.1

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
  - Deploy a TiFlash contains that commit
  - Deploy a TiDB contains this commit
  - Check the query result of `select * from information_schema.tiflash_tables limit 1`

Side effects

- Breaking backward compatibility

### Release note 

- No release note

### CODE REVIEW

File: infoschema/tables.go [hunk 1]
```diff
@@ -1271,14 +1271,20 @@ var tableTableTiFlashTablesCols = []columnInfo{
...
 	{name: "STORAGE_STABLE_MAX_PAGE_ID", tp: mysql.TypeLonglong, size: 64},
 	{name: "STORAGE_DELTA_NUM_SNAPSHOTS", tp: mysql.TypeLonglong, size: 64},
+	{name: "STORAGE_DELTA_OLDEST_SNAPSHOT_LIFETIME", tp: mysql.TypeDouble, size: 64},
+	{name: "STORAGE_DELTA_OLDEST_SNAPSHOT_THREAD_ID", tp: mysql.TypeLonglong, size: 64},
 	{name: "STORAGE_DELTA_NUM_PAGES", tp: mysql.TypeLonglong, size: 64},
 	{name: "STORAGE_DELTA_NUM_NORMAL_PAGES", tp: mysql.TypeLonglong, size: 64},
 	{name: "STORAGE_DELTA_MAX_PAGE_ID", tp: mysql.TypeLonglong, size: 64},
 	{name: "STORAGE_META_NUM_SNAPSHOTS", tp: mysql.TypeLonglong, size: 64},
+	{name: "STORAGE_META_OLDEST_SNAPSHOT_LIFETIME", tp: mysql.TypeDouble, size: 64},
+	{name: "STORAGE_META_OLDEST_SNAPSHOT_THREAD_ID", tp: mysql.TypeLonglong, size: 64},
```
Comment (JaySon-Huang) [inline infoschema/tables.go:1287]: I'm not sure whether we can add columns in the middle of it. Or we should always add columns behind all existing columns. Please let me know which is better.

Comment (lidezhu) [reply infoschema/tables.go:1287]: Both way is ok.

**Comment by JaySon-Huang** [comment]
@crazycs520 PTAL

**Review by crazycs520 — APPROVED** [review]
LGTM

File: infoschema/tables.go (hunk 1 above)
Comment (tangenta) [reply infoschema/tables.go:1287]: This may break some hard-coded script that uses an integer as the column index. But I think this is acceptable.

**Review by tangenta — APPROVED** [review]

**Pull request merged**